src/types.rs

```diff
@@ -488,6 +490,146 @@ impl<T> Line<T>
 pub struct LineString<T>(pub Vec<Point<T>>) where T: Float;
 
 impl<T: Float> LineString<T> {
+    /// Create a new `LineString` from a `Point`s vector.
+    ///
+    /// # Errors
+    ///
+    /// This function returns an `Err` if `points` has fewer than two `Point`s.
+    ///
+    /// # Examples
+    ///
+    /// Creating a `LineString`:
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![Point::new(6., 14.), Point::new(2., -10.)];
+    /// assert!(LineString::new(points).is_ok());
+    /// ```
+    ///
+    /// Attempting to create a `LineString` with fewer than two `Point`s
+    /// results in an error:
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![Point::new(6., 14.)];
+    /// assert!(LineString::new(points).is_err());
+    /// ```
+    pub fn new(points: Vec<Point<T>>) -> Result<Self, LineStringError<T>> {
```

Comment:
I feel like this might make the library painful to use. Returning a `Result` means you now need to add loads of `unwrap()` statements to your code seeing as 99% of the time you already *know* that the line you created was valid.

What about an `assert!()` so you get a runtime panic instead? I'd argue creating an invalid `LineString` is a programming error, so it should blow up loudly during development time. In production code it should be impossible to create an invalid line, and if you do then chances are your program is FUBAR anyway...

Reply:
A small detail, but since this returns a `Result`, one should be able to do `LineString::new(points)?` in many contexts